File: python/PhoenixSysId/robotcontainer.py
from commands2 import Command, cmd
from commands2.button import CommandXboxController
from commands2.sysid import SysIdRoutine

from phoenix6 import SignalLogger

from subsystems.flywheel import FlywheelMechanism

class RobotContainer:
    def __init__(self) -> None:
        self.joystick = CommandXboxController(0)
        self.mechanism = FlywheelMechanism()

        self.configure_bindings()

    def configure_bindings(self) -> None:
        """Use this method to define bindings between conditions and commands. These are useful for
        automating robot behaviors based on button and sensor input.

        Should be called during :meth:`.Robot.__init__`.

        Event binding methods are available on the :class:`.Trigger` class.
        """

        # Default command is duty cycle control with the left up/down stick
        self.mechanism.set_default_command(self.mechanism.joystick_drive_command(self.joystick.get_left_y))

        # Manually start logging with left bumper before running any tests,
        # and stop logging with right bumper after we're done with ALL tests.
        # This isn't necessary but is convenient to reduce the size of the hoot file.
        self.joystick.left_bumper().on_true(cmd.run_once(SignalLogger.start))
        self.joystick.right_bumper().on_true(cmd.run_once(SignalLogger.stop))
        
        # Joystick Y = quasistatic forward
        # Joystick A = quasistatic reverse
        # Joystick B = dynamic forward
        # Joystick X = dynamic reverse
        self.joystick.y().while_true(self.mechanism.sys_id_quasistatic(SysIdRoutine.Direction.FORWARD))
        self.joystick.a().while_true(self.mechanism.sys_id_quasistatic(SysIdRoutine.Direction.REVERSE))
        self.joystick.b().while_true(self.mechanism.sys_id_dynamic(SysIdRoutine.Direction.FORWARD))
        self.joystick.x().while_true(self.mechanism.sys_id_dynamic(SysIdRoutine.Direction.REVERSE))

    def get_autonomous_command(self) -> Command:
        """Use this to define the command that runs during autonomous.

        Scheduled during :meth:`Robot.autonomous_init`.
        """

        return cmd.print_("No autonomous command configured")

File: cpp/PhoenixSysId/src/main/cpp/RobotContainer.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "RobotContainer.hpp"

#include "wpi/commands2/Commands.hpp"

using namespace ctre::phoenix6;

RobotContainer::RobotContainer()
{
    ConfigureBindings();
}

void RobotContainer::ConfigureBindings()
{
    /* Default command is duty cycle control with the left up/down stick */
    m_mechanism.SetDefaultCommand(m_mechanism.JoystickDriveCommand([this] { return m_joystick.GetLeftY(); }));

    /* Manually start logging with left bumper before running any tests,
     * and stop logging with right bumper after we're done with ALL tests.
     * This isn't necessary but is convenient to reduce the size of the hoot file.*/
    m_joystick.LeftBumper().OnTrue(wpi::cmd::RunOnce(SignalLogger::Start));
    m_joystick.RightBumper().OnTrue(wpi::cmd::RunOnce(SignalLogger::Stop));

    /*
     * Joystick Y = quasistatic forward
     * Joystick A = quasistatic reverse
     * Joystick B = dynamic forward
     * Joystick X = dyanmic reverse
     */
    m_joystick.Y().WhileTrue(m_mechanism.SysIdQuasistatic(wpi::cmd::sysid::Direction::kForward));
    m_joystick.A().WhileTrue(m_mechanism.SysIdQuasistatic(wpi::cmd::sysid::Direction::kReverse));
    m_joystick.B().WhileTrue(m_mechanism.SysIdDynamic(wpi::cmd::sysid::Direction::kForward));
    m_joystick.X().WhileTrue(m_mechanism.SysIdDynamic(wpi::cmd::sysid::Direction::kReverse));
}

wpi::cmd::CommandPtr RobotContainer::GetAutonomousCommand()
{
    return wpi::cmd::Print("No autonomous command configured");
}
